lib/oli_web/live/resources/pages_view.ex

```diff
@@ -272,6 +285,167 @@ defmodule OliWeb.Resources.PagesView do
+      from_container: from_container,
+      active: active
+    }
+
+    {:noreply,
+     assign(socket,
+       modal: %{component: MoveModal, assigns: assigns}
+     )}
+  end
+
+  def handle_event(
+        "MoveModal.move_item",
+        %{"from_uuid" => from_uuid, "to_uuid" => to_uuid},
+        socket
+      ) do
+    %{
+      author: author,
+      project: project,
+      modal: %{assigns: %{node: node, hierarchy: hierarchy}}
+    } = socket.assigns
+
+    %{revision: revision} = node
+
+    from_container =
+      case Hierarchy.find_parent_in_hierarchy(hierarchy, from_uuid) do
+        %{revision: from_container} -> from_container
+        _ -> nil
+      end
+
+    %{revision: to_container} = Hierarchy.find_in_hierarchy(hierarchy, to_uuid)
+
+    {:ok, _} = ContainerEditor.move_to(revision, from_container, to_container, author, project)
+
+    {:noreply, hide_modal(socket)}
+  end
+
+  def handle_event("MoveModal.remove", %{"from_uuid" => from_uuid}, socket) do
+    %{
+      author: author,
+      project: project,
+      modal: %{assigns: %{node: node, hierarchy: hierarchy}}
+    } = socket.assigns
+
+    %{revision: revision} = node
+    %{revision: from_container} = Hierarchy.find_in_hierarchy(hierarchy, from_uuid)
+    to_container = nil
+
+    {:ok, _} = ContainerEditor.move_to(revision, from_container, to_container, author, project)
+
+    {:noreply, hide_modal(socket)}
+  end
+
+  def handle_event("MoveModal.cancel", _, socket) do
+    {:noreply, hide_modal(socket)}
+  end
+
+  def handle_event("HierarchyPicker.update_active", %{"uuid" => uuid}, socket) do
+    %{modal: %{assigns: %{hierarchy: hierarchy}} = modal} = socket.assigns
+
+    active = Hierarchy.find_in_hierarchy(hierarchy, uuid)
+
+    modal = %{
+      modal
+      | assigns: %{
+          modal.assigns
+          | active: active
+        }
+    }
+
+    {:noreply, assign(socket, modal: modal)}
+  end
+
+  # handle clicking of the "Add Graded Assessment" or "Add Practice Page" buttons
+  def handle_event("create_page", _, socket) do
+    %{
+      project: project,
+      author: author,
+      table_model: table_model,
+      options: options,
+      offset: offset
+    } = socket.assigns
+
+    attrs = %{
+      tags: [],
+      objectives: %{"attached" => []},
+      children: [],
+      content: %{
+        "version" => "0.1.0",
+        "model" => []
+      },
+      title: "New Page",
+      graded: false,
+      max_attempts: 0,
+      recommended_attempts: 0,
+      scoring_strategy_id: Oli.Resources.ScoringStrategy.get_id_by_type("best"),
+      resource_type_id: Oli.Resources.ResourceType.get_id_by_type("page")
+    }
+
+    socket =
+      case ContainerEditor.add_new(
+             nil,
+             attrs,
+             author,
+             project
+           ) do
+        {:ok, _} ->
+          socket
+
+        {:error, %Ecto.Changeset{} = _changeset} ->
+          socket
+          |> put_flash(:error, "Could not create new page")
+      end
```

Comment:
After I clicked the button to create a page (in the Chemistry course, which has 500 pages), I saw nothing happen.  No updates, no modal.  So I clicked again, and again.  Then, I realized it was creating the pages, but there were just not in my current "view" of the 25 rows of pages. 

I think a better UX would be to take the user into the "page editor" for the new page immediately after it is created.  I suspect the common use case here is "I am creating a new page because I want to author a new page"

Reply:
Good point, I agree. I will add this in

Reply:
Added the ability to create practice, graded and adaptive (when enabled), same as container live view.
<img width="335" alt="Screen Shot 2022-10-04 at 2 01 09 PM" src="https://user-images.githubusercontent.com/6248894/193895352-204ede9f-1a4c-44ca-9e53-eeaed0b6b9bd.png">